### PR TITLE
ATT-28: Enable 2.1.x and 2.2.x Support

### PR DESCRIPTION
**Pull request by chinzou**
https://issues.openmrs.org/browse/ATT-28

### CODE REVIEW

File: omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java [hunk 1]
```diff
@@ -84,4 +95,32 @@ public void postAttachment_shouldUpdateObsComment() throws Exception {
...
+		AttachmentsService as = Context.getService(AttachmentsService.class);
+		List<Attachment> attachments = as.getAttachments(patient, false);
+		
+		// Replay
+		MockHttpServletRequest request = newGetRequest(getURI() + "/");
+		request.addParameter("patient", patient.getUuid());
+		request.addParameter("v", "full");
+		
+		RequestContext requestContext = new RequestContext();
+		requestContext.setRequest(request);
```
Comment (mks-d) [inline omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java:114]: All the above is actually still the _setup_.

File: omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java [hunk 2]
```diff
@@ -84,4 +95,32 @@ public void postAttachment_shouldUpdateObsComment() throws Exception {
...
+		PageableResult response = res.doSearch(requestContext);
+		
+		BasePageableResult basePageableResult = (BasePageableResult) response;
```
Comment (mks-d) [inline omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java]: That's a detail but if possible: `basePageableResult` → `pageableResult`.

File: omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java [hunk 3]
```diff
@@ -84,4 +95,32 @@ public void postAttachment_shouldUpdateObsComment() throws Exception {
...
+		List results = basePageableResult.getPageOfResults();
```
Comment (mks-d) [inline omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java]: Extract the list of `Attachment` here already:
```java
List<Attachment> actualAttachments = (List<Attachment>) pageableResult.getPageOfResults();
```

File: omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java [hunk 4]
```diff
@@ -84,4 +95,32 @@ public void postAttachment_shouldUpdateObsComment() throws Exception {
...
+	
+	@Test
+	public void doSearch_shouldReturnResult() {
+		
+		// Setup
+		AttachmentResource1_10 res = new AttachmentResource1_10();
+		Patient patient = Context.getPatientService().getPatient(2);
+		
+		AttachmentsService as = Context.getService(AttachmentsService.class);
+		List<Attachment> attachments = as.getAttachments(patient, false);
```
Comment (mks-d) [inline omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java:107]: `attachments` → `actualAttachments`

**Review by mks-d — COMMENTED** [review]
That's great!

I left a few nitpicking comments about renaming/reorganising the unit test.

File: omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java (hunk 2 above)
Comment (chinzou) [reply omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java]: pageableResult doesn't return a list of results whereas BasePageableResult do, however an improvement is possible

File: omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java (hunk 3 above)
Comment (chinzou) [reply omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java]: Yes

File: omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java (hunk 4 above)
Comment (chinzou) [reply omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentController2_0Test.java:107]: Ok ;)

**Pull request merged**